pkg/inject/inject.go

```diff
@@ -263,8 +275,14 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 			return err
 		}
 
-		if v.Name == controllerPodName && v.Namespace == conf.configs.GetGlobal().GetLinkerdNamespace() {
-			conf.destinationDNSOverride = localhostDNSOverride
+		if v.Namespace == conf.configs.GetGlobal().GetLinkerdNamespace() {
+			switch v.Name {
+			case controllerDeployName:
+				conf.destinationDNSOverride = localhostDNSOverride
+			case identityDeployName:
+				conf.identityDNSOverride = localhostDNSOverride
+			default:
```

Comment:
No need for the empty default on the switch statement here -- you can remove this line.